### PR TITLE
Patch for push name

### PR DESCRIPTION
**Pull request by CODeRUS**
setting own push name in profile settings
setting push name for unknown contacts on new message

### CODE REVIEW

**Comment by CODeRUS** [comment]
please test and comment here

File: src/client/contacts.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def updateContacts(self,data):
 					contact.iscontact = "yes"
 					contact.save()
 
-			self.contactsRefreshSuccess.emit(self.mode, "");	
+			self.contactsRefreshSuccess.emit(self.mode, []);
```
Comment (tgalal) [inline src/client/contacts.py]: this signature is QtCore.Signal(str,dict); 
so this should be 
self.contactsRefreshSuccess.emit(self.mode, {} );

I'll update it.

File: src/client/ui.py [hunk 2]
```diff
@@ -45,7 +46,7 @@ class WAUI(QDeclarativeView):
 	phoneContactsReady = QtCore.Signal(list)
 
 	
-	def __init__(self, accountJid):
+	def __init__(self, accountJid, accountPushName):
```
Comment (tgalal) [inline src/client/ui.py:49]: accountPushName is not necessary to construct the UI, so I'll set it from a separate function

File: src/client/waxmpp.py [hunk 3]
```diff
@@ -920,9 +937,9 @@ def onUnavailable(self):
 	def conversationOpened(self,jid):
 		self.notifier.hideNotification(jid);
 	
-	def onAvailable(self):
+	def onAvailable(self, pushname):
```
Comment (tgalal) [inline src/client/waxmpp.py:940]: presence_sendAvailable is still a valid stanza, so to remain accessible I'll make pushname an  optional parameter, if provided then presence_sendAvailableForChat, if not then presence_sendAvailable

**Comment by tgalal** [comment]
All working great, thanks!

**Comment by CODeRUS** [comment]
uff, thanks god it working :D

**Comment by tgalal** [comment]
just noticed pushname is not updated for unknown contacts in UI :P I'm investigating the reason now

**Comment by CODeRUS** [comment]
i tested last tonight with brkn, it worked =)

**Comment by tgalal** [comment]
Now I see it works, but, the search loop is expensive, it takes time and blocks the UI for a second or 2

**Comment by tgalal** [comment]
Why send all contacts when you only need to update a single contact's pushname?

**Comment by tgalal** [comment]
in:

def updatePushName(self, jid, push):
        self._d("UPDATING CONTACTS");
        contacts = self.c.getContacts();
        self.rootObject().updateContactsData(contacts, jid, push);

**Comment by CODeRUS** [comment]
@cepiperez =)

**Comment by tgalal** [comment]
as usual..

**Comment by CODeRUS** [comment]
i just extracted changes with kdiff3 and made some code cleanup

**Comment by tgalal** [comment]
Fixed it. 
@cepiperez, when will you start adding any quality to what you do ?

**Comment by CepiPerez** [comment]
sorry for that
next time I'll pull a decent job... if I get the time :P

**Pull request merged**